pkg/diagnostics/grpc_tracing.go

```diff
@@ -51,13 +51,31 @@ func GRPCTraceUnaryServerInterceptor(appID string, spec config.TracingSpec) grpc
 			spanKind = trace.WithSpanKind(trace.SpanKindClient)
 		}
 
-		ctx, span = trace.StartSpanWithRemoteParent(ctx, spanName, sc, sampler, spanKind)
+		ctx, span = trace.StartSpanWithRemoteParent(ctx, info.FullMethod, sc, sampler, spanKind)
+		if span.SpanContext().IsSampled() {
```

Comment:
Please, use a variable for this and reuse the same variable in the following `if`, this way both `ifs` are either on or off and not mixed.